gobblin-modules/gobblin-elasticsearch/src/test/java/org/apache/gobblin/elasticsearch/ElasticsearchTestServer.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gobblin.elasticsearch;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Arrays;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import com.google.common.base.Throwables;
+
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * A Test ElasticSearch server
+ */
+@Slf4j
+public class ElasticsearchTestServer {
+
+
+  private static final String ELASTICSEARCH_VERSION="5.6.8";
+  private static final String ELASTICSEARCH_PATH_PREFIX="gobblin-modules/gobblin-elasticsearch/test-elasticsearch/";
+  private static final String ELASTICSEARCH_BIN=ELASTICSEARCH_PATH_PREFIX + "elasticsearch-"
+      + ELASTICSEARCH_VERSION + "/bin/elasticsearch";
+
+
+  private Process elasticProcess;
+  private int _port = 9200;
```

Comment:
how about using a free port available at the time, like: https://stackoverflow.com/questions/51099027/find-free-port-in-java 
we had frequent transient test failures in service / cluster test cases when they were using fixed ports to setup pre-requisite infra to run tests